### DISCUSSION TITLE
Sporadic 500 Errors on SharePoint Online PnP search

### DISCUSSION THREAD

**Discussion by pgeorgieva87** [microsoft-search/pnp-modern-search#3346] [Q&A]
![image](https://github.com/microsoft-search/pnp-modern-search/assets/151198751/5f542bf5-4f09-4758-ab79-3663f89ae676)

Seeing those intermittent 500 errors on SharePoint online landing page. 

This is a  landing/ home page that contains PnP search links, and sometimes (sporadically) shows 500 Error (CorrelationId) errors. 

Then it will go back to normal with no change on our side. 

After a few weeks someone will report as issue again...

Any ideas what could be happening here ?

**Comment by gretchunkim** [comment]
@pgeorgieva87  Is this during business hours or off hours? I have seen something like this myself, and for me was due to throttling... If the issue is resolved in let's say 15 minutes for  one user, then just let users to try again in 15-30 minutes. 
If this is happening to more users (not just another users but at the same timeframe) then open a ticket with Microsoft with the correlation ID and they will be able to look up backend services information for you.

**Comment by pgeorgieva87** [comment]
It is only during business hours. We opened up a ticket with Microsoft and they told us to reach out to the developers of PNP.

**Reply by wobba** [reply]
Any API issue, you have to replicate it outside of these web parts to get support. Sad, but true. This is an open source solution and we have no control of the API. If our API request is somewhat wrong, then we can address it. But we have not changed anything in a long time, and if it's intermittent, then it has to be looked at from the API side of things.

**Comment by stue-he** [comment]
We are experiencing the same issue and received the same feedback from Microsoft Support.
I was hoping an update of the pnp search to version 4.10.1 (currently used is 4.8.0.0) could help solve this.

If I interpret the answers to this post correctly it probably wouldn't help at all, would it?

**Reply by kasperbolarsen** [reply]
Not likely, no. 
However, if you have the skills inhouse, you might be able to setup a script that calls the API once every x minutes, and once the API is throwing the 500 error you will have the proof that shows MS that the issue has nothing to do with PnP Modern Search. If you are lucky MS support *might* be able to provide this debugging script .
